crates/compiler/mono/src/ir.rs

```diff
@@ -7799,6 +8021,68 @@ fn store_newtype_pattern<'a>(
     }
 }
 
+#[allow(clippy::too_many_arguments)]
+fn store_tuple_destruct<'a>(
+    env: &mut Env<'a, '_>,
+    procs: &mut Procs<'a>,
+    layout_cache: &mut LayoutCache<'a>,
+    destruct: &TupleDestruct<'a>,
+    index: u64,
+    outer_symbol: Symbol,
+    sorted_fields: &'a [InLayout<'a>],
+    mut stmt: Stmt<'a>,
+) -> StorePattern<'a> {
+    use Pattern::*;
+
+    let load = Expr::StructAtIndex {
+        index,
+        field_layouts: sorted_fields,
+        structure: outer_symbol,
+    };
+
+    match &destruct.pat {
+        Identifier(symbol) => {
+            stmt = Stmt::Let(*symbol, load, destruct.layout, env.arena.alloc(stmt));
+        }
+        Underscore => {
+            // important that this is special-cased to do nothing: mono record patterns will extract all the
+            // fields, but those not bound in the source code are guarded with the underscore
+            // pattern. So given some record `{ x : a, y : b }`, a match
+            //
+            // { x } -> ...
+            //
+            // is actually
+            //
+            // { x, y: _ } -> ...
+            //
+            // internally. But `y` is never used, so we must make sure it't not stored/loaded.
```

Comment:
This comment should be updated for tuples.

Reply:
This was copy+pasta, and on further reading, it doesn't really say anything helpful. Gone.